chart/prometheus-msteams/values.yaml

```diff
@@ -45,7 +45,22 @@ metrics:
     scrapeInterval: 30s
 
 connectors: []
-connectors_with_custom_templates: []
+
+# Templates defined here can be used in connectorsWithCustomTemplates and are mounted into /etc/template/
+customCardTemplates: []
+# - templateFileName: custom-message-card.tmpl
+#   content: |
+#     {{ define "teams.card" }}
+#     {...}
+#     {{ end }}
+
+# ref: https://github.com/prometheus-msteams/prometheus-msteams#customise-messages-per-ms-teams-channel
+# Templates are mounted into /etc/template/
+connectorsWithCustomTemplates: []
+# - request_path: /alert2
+#   template_file: /etc/template/custom-message-card.tmpl
+#   webhook_url: <webhook>
+#   escape_underscores: true
 
```

Comment:
In order for this to work, we need a config.yaml like this:

```yaml
customCardTemplates:
  - templateFileName: card1.tmpl
  - templateFileName: card2.tmpl

connectorsWithCustomTemplates:
  - request_path: /alert1
    template_file: /etc/template/card1.tmpl
    webhook_url: https://outlook.office.com/webhook/xxxx
    escape_underscores: true
  - request_path: /alert2
    template_file: /etc/template/card2.tmpl
    webhook_url: https://outlook.office.com/webhook/yyyy
    escape_underscores: true
```

What I don't like in this approach is, that you need to provide both `customCardTemplates` and `connectorsWithCustomTemplates`. Additionally with `template_file: /etc/template/card1.tmpl` the user always has to provide `/etc/template/` (this shouldn't be the users concern), and the `card1.tmpl` in  /etc/template/card1.tmpl` has to match the `- templateFileName: card1.tmpl` in `customCardTemplates`.

How about we make this more easy: what if the user simply provides:

```yaml
connectorsWithCustomTemplates:
  - request_path: /alert1
    webhook_url: https://outlook.office.com/webhook/xxxx
    escape_underscores: true
  - request_path: /alert2
    webhook_url: https://outlook.office.com/webhook/yyyy
    escape_underscores: true
```

where `template_file` then can be provided via `--set-file`, i.e. with

```shell
helm upgrade --install prometheus-msteams \
  --namespace default -f ~/.prometheus-msteams/helm-config.yaml \
  --set-file connectorsWithCustomTemplates\[0\].template_file=./default-message-card.tmpl \
  --set-file connectorsWithCustomTemplates\[1\].template_file=./examples/templates/card-with-action.tmpl \
  ./chart/prometheus-msteams
```

A second option is to provide `template_file` inline like

```yaml
connectorsWithCustomTemplates:
  - request_path: /alert1
    webhook_url: https://outlook.office.com/webhook/xxxx
    escape_underscores: true
    template_file: |
      {{ define "teams.card" }}
      ...
      {{ end }}
```

and then we completely get rid of `customCardTemplates`. What do you think?

Reply:
Good point, this approach looks much cleaner.

Reply:
Good point, this approach looks much cleaner.